core/src/main/scala/cats/evidence/As.scala

```diff
@@ -0,0 +1,268 @@
+    a.subst[λ[`-α` => T[A, Z, C] <~< T[A, α, C]]](refl)
+
+  def contra3_3[T[_, _, -_], Z, A, B, C](a: C <~< Z): (T[A, B, Z] <~< T[A, B, C]) =
+    a.subst[λ[`-α` => T[A, B, Z] <~< T[A, B, α]]](refl)
+
+  def contra1_4[T[-_, _, _, _], Z, A, B, C, D](a: A <~< Z): (T[Z, B, C, D] <~< T[A, B, C, D]) =
+    a.subst[λ[`-α` => T[Z, B, C, D] <~< T[α, B, C, D]]](refl)
+
+  def contra2_4[T[_, -_, _, _], Z, A, B, C, D](a: B <~< Z): (T[A, Z, C, D] <~< T[A, B, C, D]) =
+    a.subst[λ[`-α` => T[A, Z, C, D] <~< T[A, α, C, D]]](refl)
+
+  def contra3_4[T[_, _, -_, _], Z, A, B, C, D](a: C <~< Z): (T[A, B, Z, D] <~< T[A, B, C, D]) =
+    a.subst[λ[`-α` => T[A, B, Z, D] <~< T[A, B, α, D]]](refl)
+
+  def contra4_4[T[_, _, _, -_], Z, A, B, C, D](a: D <~< Z): (T[A, B, C, Z] <~< T[A, B, C, D]) =
+    a.subst[λ[`-α` => T[A, B, C, Z] <~< T[A, B, C, α]]](refl)
+
+  /**
+   *  Lift subtyping into a Function1-like type
+   *  liftF1(a,r) = contra1_2(a) compose co2_2(b)
+   */
+  def liftF1[F[-_, +_], A, A2, R, R2](
+    a: A <~< A2,
+    r: R <~< R2
+  ): (F[A2, R] <~< F[A, R2]) = {
+    type a[-X] = F[A2, R2] <~< F[X, R2]
+    type r[-X] = F[A2, X] <~< F[A, R2]
+    r.subst[r](a.subst[a](refl))
+  }
+
+  /**
+   * Lift subtyping into a function
+   */
+  def liftF2[F[-_, -_, +_], A, A2, B, B2, R, R2](
+    a: A <~< A2,
+    b: B <~< B2,
+    r: R <~< R2
+  ): (F[A2, B2, R] <~< F[A, B, R2]) = {
+    type a[-X] = F[A2, B2, R2] <~< F[X, B2, R2]
+    type b[-X] = F[A2, B2, R2] <~< F[A, X, R2]
+    type r[-X] = F[A2, B2, X] <~< F[A, B, R2]
+    r.subst[r](b.subst[b](a.subst[a](refl)))
+  }
+
+  /**
+   * Lift subtyping into a function
+   * liftF3(a,b,c,r) = contra1_4(a) compose contra2_4(b) compose contra3_4(c) compose co3_4(d)
+   */
+  def liftF3[F[-_, -_, -_, +_], A, A2, B, B2, C, C2, R, R2](
+     a: A <~< A2,
+     b: B <~< B2,
+     c: C <~< C2,
+     r: R <~< R2
+  ): (F[A2, B2, C2, R] <~< F[A, B, C, R2]) = {
+    type a[-X] = F[A2, B2, C2, R2] <~< F[X, B2, C2, R2]
+    type b[-X] = F[A2, B2, C2, R2] <~< F[A, X, C2, R2]
+    type c[-X] = F[A2, B2, C2, R2] <~< F[A, B, X, R2]
+    type r[-X] = F[A2, B2, C2, X] <~< F[A, B, C, R2]
+    r.subst[r](c.subst[c](b.subst[b](a.subst[a](refl))))
+  }
+
+  /**
+   * Lift subtyping into a function 
+   */
+  def liftF4[F[-_, -_, -_, -_, +_], A, A2, B, B2, C, C2, D, D2, R, R2](
+     a: A <~< A2,
+     b: B <~< B2,
+     c: C <~< C2,
+     d: D <~< D2,
+     r: R <~< R2
+   ): (F[A2, B2, C2, D2, R] <~< F[A, B, C, D, R2]) = {
+    type a[-X] = F[A2, B2, C2, D2, R2] <~< F[X, B2, C2, D2, R2]
+    type b[-X] = F[A2, B2, C2, D2, R2] <~< F[A, X, C2, D2, R2]
+    type c[-X] = F[A2, B2, C2, D2, R2] <~< F[A, B, X, D2, R2]
+    type d[-X] = F[A2, B2, C2, D2, R2] <~< F[A, B, C, X, R2]
+    type r[-X] = F[A2, B2, C2, D2, X] <~< F[A, B, C, D, R2]
+    r.subst[r](d.subst[d](c.subst[c](b.subst[b](a.subst[a](refl)))))
+  }
+
+  /**
+   * Unsafely force a claim that A is a subtype of B 
+   */
+  def force[A, B]: A <~< B =
```

Comment:
I don't think we should have this. You can call isa and cast if you want to do something unsafe. The cast at least shows you it is unsafe. Burying an unsafe cast in a library function seems like a bad choice to me.